### PR TITLE
Adding week 2 answers and explanations to the markdown file

### PR DESCRIPTION
**Pull request by dmoracze**

### CODE REVIEW

File: week_2/week_2_quiz.md [hunk 1]
```diff
@@ -40,7 +62,29 @@ Select all that apply:
...
+
+`echo $SHELL`
+
+`echo $SHELL > shell.txt | cat shell.txt`
+
+**Note**
+
+`echo $SHELL > shell.txt | cat shell.txt` is an intentionally convuluted and unnecessary solution meant to explain shell concepts.
+
+**Explaination**
```
Comment (jdkent) [inline week_2/week_2_quiz.md:76]: typo
```suggestion
**Explanation**
```

File: week_2/week_2_quiz.md [hunk 2]
```diff
@@ -50,22 +94,89 @@ Hint: you may need to use pipes.
...
+
+Your answer will depend on your operating system. The answer of 28 lines was found using Mac OS 10.14.6.
+
+**Possible solutions**
+
+`man grep | grep pattern | wc -l`
+
+`man grep | grep -c pattern`
+
+**Explaination**
```
Comment (jdkent) [inline week_2/week_2_quiz.md:112]:
```suggestion
**Explanation**
```

File: week_2/week_2_quiz.md [hunk 3]
```diff
@@ -50,22 +94,89 @@ Hint: you may need to use pipes.
...
+
+Your answer will depend on your operating system. The answer of `cat.txt`, `grep.txt`, `ls.txt` was found using Mac OS 10.14.6.
+
+**Possible solutions**
+
+`wc -l *.txt | sort`
+
+`wc -l *.txt > lines.txt | sort lines.txt`
+
+**Explaination**
```
Comment (jdkent) [inline week_2/week_2_quiz.md:146]:
```suggestion
**Explanation**
```

File: week_2/week_2_quiz.md [hunk 4]
```diff
@@ -50,22 +94,89 @@ Hint: you may need to use pipes.
...
 - `mv grep.txt cat.txt ls.txt user_manuals/`
 
+<details>
+<summary>Click to see answer</summary>
+
+`mv *.txt user_manuals/`
+
+`mv grep.txt cat.txt ls.txt user_manuals/`
+
+**Explaination**
```
Comment (jdkent) [inline week_2/week_2_quiz.md:172]:
```suggestion
**Explanation**
```

File: week_2/week_2_quiz.md [hunk 5]
```diff
@@ -88,20 +200,97 @@ After the commit, what is the status of the repository?
...
+
+master
+
+**Solution**
+
+Create new branch: `git branch new_feature`
+
+Check the status `git status` and see that you're still on the master branch.
+
+**Explaination**
```
Comment (jdkent) [inline week_2/week_2_quiz.md:242]:
```suggestion
**Explanation**
```

File: week_2/week_2_quiz.md [hunk 6]
```diff
@@ -88,20 +200,97 @@ After the commit, what is the status of the repository?
...
+
+**Solution**
+
+Checkout `new_feature`: `git checkout new_feature`
+
+Create new file: `echo code for new feature > awesome_feature.txt`
+
+Add to staging area: `git add awesome_feature.txt`
+
+Commit: `git commit -m "adding awesome feature`
```
Comment (jdkent) [inline week_2/week_2_quiz.md:276]: missing quotation
```suggestion
Commit: `git commit -m "adding awesome feature"`
```

File: week_2/week_2_quiz.md [hunk 7]
```diff
@@ -88,20 +200,97 @@ After the commit, what is the status of the repository?
...
+Count number of lines in log: `git log | wc -l`
+
+Your log should look something like (with a different commit ID):
+
+```
+commit 63a47a48c7673f87b7c8ab463fa766dee3b485e7 (HEAD -> master)
+Author: user.name <user.email>
+Date:   DATE
+
+    added grep, cat, and ls manuals
```
Comment (jdkent) [inline week_2/week_2_quiz.md:291]: imperative commit message.
```suggestion
    add grep, cat, and ls manuals
```

**Review by jdkent — COMMENTED** [review]
Love the usage of the drop down buttons, I have couple nitpicks about spelling and commit messages.

**Comment by dmoracze** [comment]
@jdkent thanks for the review, should be good now

**Review by jdkent — APPROVED** [review]
LGTM!

**Pull request merged**